### PR TITLE
Skip tls config when passthrough is enabled

### LINKED ISSUE #425: Skip tls config when passthrough is enabled
### Bug Description

When creating a tcp route via the `submit_to_traefik` method in the charm lib, two routes are created instead of one and the second overrides the first. For instance:
```
juju-stg-wazuh-k8s-wazuh-server-enrole-tcp:
      entryPoints:
      - enrole-tcp
      rule: HostSNI(`*`)
      service: juju-stg-wazuh-k8s-wazuh-server-service-enrole-tcp
      tls:
        passthrough: true
```
results in
```
    juju-stg-wazuh-k8s-wazuh-server-enrole-tcp:
      entryPoints:
      - enrole-tcp
      rule: HostSNI(`*`)
      service: juju-stg-wazuh-k8s-wazuh-server-service-enrole-tcp
      tls:
        passthrough: true
    juju-stg-wazuh-k8s-wazuh-server-enrole-tcp-tls:
      entryPoints:
      - enrole-tcp
      rule: HostSNI(`*`)
      service: juju-stg-wazuh-k8s-wazuh-server-service-enrole-tcp
      tls: {}
```
being the second route name larger, it is matched, rendering the `tls` config useless.

### To Reproduce

Related a charm that configures traefik as follows https://github.com/canonical/wazuh-server-operator/blob/main/src/traefik_route_observer.py#L99
Probably simpler to code a failing test than actually reproduce with a Wazuh deployment, which requires several of components

### Environment

rev 194

### Relevant log output

```shell
None
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by IbraAoad**
## Issue
Fixes #425 

## Solution
Skip the generation of tls dynamic config in traffic when passthrough is set

## Context
When `tls.passthrough: true` is set for a route, Traefik forwards the raw TLS traffic directly to the backend service without terminating the TLS connection.
This means Traefik doesn't need to handle certificates, nor does it need additional TLS configuration 

## Testing Instructions
1- juju deploy traefik-k8s 
2- juju deploy [wazuh-server-operator](https://github.com/canonical/wazuh-server-operator)
3- juju relate traefik-k8s wazuh-server-operator
4- Fetch traefik's dynamic config and validate no tls entry is created and main entry has `tls.passthrough: true`

### CODE REVIEW

File: src/charm.py
[the remark this answers is not in the document]
Comment (ca-scribner) [reply src/charm.py:1]: the examples [here](https://doc.traefik.io/traefik/routing/routers/#tls) for `Routes for HTTP & HTTPS` include two routes with the same rule, differing only by the `tls` section, so the duplicate rules do seem to be valid.  traefik must consider the tls config in this?

[the remark this answers is not in the document]
Comment (IbraAoad) [reply src/charm.py:1]: I don't really know the rationale behind the duplicate routes, it was first introduced as part of #100 so I went with "if it works, leave it there", but what your argument makes a ton of sense tho, I tried on grafana which passes tls configs on its own and ended up with the below,

```
  routers:
    juju-trfk-test-grafana-k8s-router:
      entryPoints:
      - web
      middlewares:
      - juju-sidecar-noprefix-trfk-test-grafana-k8s
      - juju-sidecar-redir-https-trfk-test-grafana-k8s
      rule: PathPrefix(`/trfk-test-grafana-k8s`)
      service: juju-trfk-test-grafana-k8s-service
    juju-trfk-test-grafana-k8s-router-tls:
      entryPoints:
      - websecure
      rule: PathPrefix(`/trfk-test-grafana-k8s`)
      service: juju-trfk-test-grafana-k8s-service
      tls:
        domains:
        - main: foo.bar
          sans:
          - '*.foo.bar'
    juju-trfk-test-grafana-k8s-router-tls-tls:
      entryPoints:
      - websecure
      rule: PathPrefix(`/trfk-test-grafana-k8s`)
      service: juju-trfk-test-grafana-k8s-service
      tls:
        domains:
        - main: foo.bar
          sans:
          - '*.foo.bar'
```

the duplicate tls route is redundant in this case, yet tbh I'm not entirely sure if anything breaks on `plaintext` communication if the added tls route isn't there

[the remark this answers is not in the document]
Comment (IbraAoad) [reply src/charm.py:1]: We can open up a new issue to investigate this, wdyt?

[the remark this answers is not in the document]
Comment (dstathis) [reply src/charm.py:1]: I think a new issue is appropriate. We have a need for this fix so we should merge it and figure out if there is an issue later.

**Review by dstathis — APPROVED** [review]

**Review by ca-scribner — APPROVED** [review]
lgtm, with the discussed complexities to be broken into a new issue

**Pull request merged**